### PR TITLE
Fix broken monaco code completion

### PR DESCRIPTION
**Pull request by pjlast**
Code completion in the site config on dotcom has been broken for a while now. The monaco editor depends on some worker scripts being available at `/.assets/scripts` for code completion suggestions to work in the online editor.

This is not an issue locally because we run an esbuild dev server that proxies requests to `/.assets`

This PR attempts to fix it by adding the `scripts` folder to the bazel build target for `//client/web/dist`

## Test plan

Tested locally by inspecting the output of `bazel-bin/client/web/dist` after running `bazel build //client/web/dist`, but I'm not sure how thorough this test is.

### CODE REVIEW

File: client/web/dist/scripts/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,16 @@
+load("@aspect_bazel_lib//lib:copy_to_bin.bzl", "copy_to_bin")
+
+filegroup(
+    name = "scripts",
+    srcs = glob(
+        ["**/*.*"],
+        ["BUILD.bazel"],
```
Comment (Strum355) [inline client/web/dist/scripts/BUILD.bazel:7]: Does this need to be included? Id be surprised if it does but what do I know :clueless:

File: client/web/dist/scripts/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    ),
+    visibility = ["//visibility:public"],
+)
+
+copy_to_bin(
+    name = "copy",
+    srcs = ["//client/web/dist/scripts"],
```
Comment (Strum355) [inline client/web/dist/scripts/BUILD.bazel:14]: Can be simplified to `":scripts"`

File: client/web/dist/scripts/BUILD.bazel (hunk 1 above)
Comment (pjlast) [reply client/web/dist/scripts/BUILD.bazel:7]: idk man I've tried various builds to get an iteration that might work. I just tried to copy what the `/img` directory is doing

**Review by valerybugakov — COMMENTED** [review]
Hey @Strum355, based on [the diff history](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+type:diff+buildMonaco%28&patternType=standard&sm=1&groupBy=author), the `buildMonaco` function was [introduced](https://github.com/sourcegraph/sourcegraph/commit/bf6fadfed9ddb66dac4bfd598a28b6bf522d5f70#diff-9fb5a352314757868e817dfdb8a23d523bb8331c7adeb8b385980af7f67a9591R67) into the web package quite some time ago, accompanied by a `TODO` note suggesting a switch to a more suitable alternative for production builds. Subsequently, during our transition from `webpack` to `esbuild` for production builds, this function [was not utilized](https://github.com/sourcegraph/sourcegraph/commit/e1f9a6f27f75588768f929afb05376ca6dc43859#diff-f7c1ab1acb5e753bccdd4b092b0098fa58855899e8764b68829ada2ba9cea760), leading to an unresolved bug since that change, as far as I understand. It appears that `buildMonaco` is now only triggered when the `esbuild/build.ts` script is executed — specifically for development builds outside of Bazel.

The `vite` [migration PR](https://github.com/sourcegraph/sourcegraph/pull/59219) addresses this issue by [resolving](https://github.com/sourcegraph/sourcegraph/blob/86180de04a599ce847540badeb0e5dacf116c83b/client/web/src/settings/MonacoSettingsEditor.tsx#L370) the initial `TODO` mentioned above the `buildMonaco` function. Therefore, we can potentially forego a specific fix for esbuild if the migration PR can be merged soon.

**Comment by pjlast** [comment]
Closing this in favour of #59219

**Pull request closed without merging**